Mathlib/LinearAlgebra/AffineSpace/Independent.lean

```diff
@@ -556,6 +556,71 @@ theorem affineCombination_mem_affineSpan_pair {p : ι → P} (h : AffineIndepend
   · simp only [Pi.sub_apply, sub_eq_iff_eq_add]
   · simp_all only [Pi.sub_apply, Finset.sum_sub_distrib, sub_self]
 
+/-!
+### Helper lemmas for affine span and nonemptiness
+-/
+
+/-- A proper affine subspace does not contain all points. -/
+lemma AffineSubspace.exists_not_mem_of_ne_top (S : AffineSubspace k P) (h : S ≠ ⊤) :
+    ∃ p : P, p ∉ S := by
+  have h_ne_univ : (S : Set P) ≠ Set.univ := by
+    intro h_eq
+    have h_top : S = ⊤ := SetLike.coe_injective h_eq
+    exact h h_top
+  exact (Set.ne_univ_iff_exists_notMem (S : Set P)).mp h_ne_univ
```

Comment:
I think this is true for any `SetLike` `OrderTop`, not an affine result at all, so should be stated in a more general context and go somewhere outside the affine space part of mathlib (I don't have a specific suggestion for where).

Reply:
Quite right, thanks! I put this tiny result, generalized to `SetLike OrderTop`, into [its own little PR](https://github.com/leanprover-community/mathlib4/pull/30946/).